### ISSUE TITLE
Running on multiple GPUs in parallel with PyTorch & Autogluon

### ISSUE THREAD

**Issue by BarclayII** [awslabs/autogluon#490]
Hi,

I was wondering if it is possible to run parallel hyperparameter trials with PyTorch.  I tried the following code but it can only utilize one GPU:
```python
@ag.args(
        BATCH_SIZE=ag.space.Categorical(256, 512, 1024),
        VAL_BATCH_SIZE=ag.space.Categorical(1000),
        HIDDEN_DIM=ag.space.Int(16, 512),
        LR=ag.space.Real(1e-5, 1e-3),
        NUM_NEGS=ag.space.Int(1, 10)
        )
def train(args, reporter):
    device = torch.device('cuda')
    # ...
    reporter(epoch=0, recall=best_metric)

sched = ag.scheduler.FIFOScheduler(train, resource={'num_cpus': 48, 'num_gpus': 12}, num_trials=36, reward_attr='recall', time_attr='epoch')
sched.run()
sched.join_jobs()
print(sched.get_best_config(), sched.get_best_reward())
```

Is it possible to make the scheduler to run multiple trials in parallel on separate GPUs?

Thanks.

**Comment by BarclayII** [comment]
Turns out that if I set both `num_cpus` and `num_gpus` to 1 it can parallelize jobs on all GPUs.  Is it intended?

And how could I specify which GPUs to run AutoGluon on?  It seems that AutoGluon will always use all GPUs available no matter what I set to `CUDA_VISIBLE_DEVICES`.

**Comment by Innixma** [comment]
Thanks for the info! While I'm glad that you were able to parallelize the GPU's, this sounds like a bug. @zhreshold could you look into this?

**Comment by zhreshold** [comment]
Autogluon use `CUDA_VISIBLE_DEVICES` to control the underlying resource management which is now incompatible with pytorch's latest changes. 
@BarclayII  In the meantime you can use `both num_cpus and num_gpus to 1` trick to manually manage the gpu resources until we find a way to manage it.
